File: Scripts/results/generated_code_results/Flutter/Shopping&E-commerce/eBay/ItemScreen.dart
import 'package:flutter/material.dart';
class ItemScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pop(context),
),
title: Text('物品'),
actions: [
IconButton(
icon: Icon(Icons.share),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.shopping_cart),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.more_vert),
onPressed: () {},
),
],
),
body: SingleChildScrollView(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Stack(
alignment: Alignment.topRight,
children: [
Image.asset('assets/images/template.png'),
Padding(
padding: const EdgeInsets.all(8.0),
child: Container(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
decoration: BoxDecoration(
color: Colors.white,
borderRadius: BorderRadius.circular(12),
),
child: Row(
mainAxisSize: MainAxisSize.min,
children: [
Text('1/1', style: TextStyle(color: Colors.black)),
],
),
),
),
Positioned(
bottom: 8,
right: 8,
child: Container(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
decoration: BoxDecoration(
color: Colors.white,
borderRadius: BorderRadius.circular(12),
),
child: Row(
mainAxisSize: MainAxisSize.min,
children: [
Icon(Icons.favorite_border, color: Colors.blue),
SizedBox(width: 4),
Text('65', style: TextStyle(color: Colors.blue)),
],
),
),
),
],
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Ottogi Jin Ramen Mild Noodles (Pack of 20)',
style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
),
SizedBox(height: 8),
Row(
children: [
Image.asset('assets/images/template.png', width: 24, height: 24),
SizedBox(width: 8),
Text('Wa Na Hong Oriental Supermarket (25024)'),
Icon(Icons.chat_bubble_outline),
],
),
Text('99.3% 好评率'),
SizedBox(height: 16),
Text(
'每件 £22.99',
style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold, color: Colors.green),
),
Text(
'Free 3 day postage',
style: TextStyle(color: Colors.green),
),
Text('预计在7月19日，星期五至7月20日，星期六之间送达'),
SizedBox(height: 16),
Text('www.WaNaHong.co.uk'),
Text('Packing: 120g x 20'),
Text('NET WT: 120g'),
],
),
),
],
),
),
bottomNavigationBar: BottomNavigationBar(
type: BottomNavigationBarType.fixed,
currentIndex: 0,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: '主页'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: '我的eBay'),
BottomNavigationBarItem(icon: Icon(Icons.search), label: '搜索'),
BottomNavigationBarItem(icon: Icon(Icons.notifications), label: '通知'),
BottomNavigationBarItem(icon: Icon(Icons.sell), label: '出售物品'),
],
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Home');
}
},
),
);
}
}

File: Scripts/results/generated_code_results/baseline_gpt/Shopping&E-commerce/eBay/ItemScreen.js
import React from 'react';
import { View, Text, Image, StyleSheet, ScrollView, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const ItemScreen = () => {
  return (
    <ScrollView style={styles.container}>
      <View style={styles.header}>
        <Icon name="chevron-back" size={30} />
        <Text style={styles.headerTitle}>商品</Text>
        <View style={styles.headerIcons}>
          <Icon name="share-social-outline" size={24} style={styles.icon} />
          <Icon name="cart-outline" size={24} style={styles.icon} />
          <Icon name="ellipsis-horizontal" size={24} style={styles.icon} />
        </View>
      </View>
      
      <View style={styles.imageContainer}>
        <Image source={require('../assets/snack-icon.png')} style={styles.image} />
        <View style={styles.likeContainer}>
          <Text style={styles.likeText}>65</Text>
          <Icon name="heart-outline" size={20} color="blue" />
        </View>
      </View>

      <View style={styles.detailsContainer}>
        <Text style={styles.title}>Ottogi Jin Ramen Mild Noodles (Pack of 20)</Text>

        <View style={styles.storeContainer}>
          <Image source={require('../assets/snack-icon.png')} style={styles.storeImage} />
          <View style={styles.storeDetails}>
            <Text style={styles.storeName}>Wa Na Hong Oriental Supermarket (25024)</Text>
            <Text style={styles.storeRating}>99.3% 好评率</Text>
          </View>
        </View>

        <Text style={styles.price}>每件 £22.99</Text>
        <Text style={styles.postage}>Free 3 day postage</Text>
        <Text style={styles.delivery}>预计在7月19日，星期五至7月20日，星期六之间送达</Text>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    padding: 10,
  },
  headerTitle: {
    fontSize: 18,
    fontWeight: '600',
  },
  headerIcons: {
    flexDirection: 'row',
  },
  icon: {
    marginLeft: 10,
  },
  imageContainer: {
    marginTop: 10,
    alignItems: 'center',
  },
  image: {
    width: '100%',
    height: 200,
    resizeMode: 'contain',
  },
  likeContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    position: 'absolute',
    bottom: 10,
    right: 10,
    backgroundColor: 'white',
    borderRadius: 15,
    padding: 5,
  },
  likeText: {
    marginRight: 5,
  },
  detailsContainer: {
    padding: 10,
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  storeContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  storeImage: {
    width: 40,
    height: 40,
    borderRadius: 20,
    marginRight: 10,
  },
  storeDetails: {
    flex: 1,
  },
  storeName: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  storeRating: {
    fontSize: 14,
    color: 'gray',
  },
  price: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  postage: {
    fontSize: 16,
    color: 'green',
    marginBottom: 5,
  },
  delivery: {
    fontSize: 16,
    color: 'gray',
  },
});

export default ItemScreen;